### PR TITLE
Fix storage index

### PR DESCRIPTION
**Pull request by cangfengzhs**
1. modify plan that contain filter.Filter node's input must be vertex/edge node install of scan node.
2. Fix some error in index unit test.

This can only solve the problem of storing and returning truncated data. It cannot deal with the problem of incorrect data set returned due to index truncation. Later, a detailed design for the processing of truncated indexes will be given for everyone to discuss

### CODE REVIEW

File: src/storage/exec/IndexOutputNode.h [hunk 1]
```diff
@@ -144,26 +150,22 @@ class IndexOutputNode final : public RelNode<T> {
         ret = vertexRowsFromData(data);
         break;
       }
+      default:
+        assert(false);
```
Comment (yixinglu) [inline src/storage/exec/IndexOutputNode.h:154]: replace the `assert` with `LOG(FATAL)`

File: src/storage/exec/IndexOutputNode.h [hunk 2]
```diff
@@ -329,9 +334,40 @@ class IndexOutputNode final : public RelNode<T> {
       default: {
         auto v = IndexKeyUtils::getValueFromIndexKey(
             context_->vIdLen(), data.first, col, fields_, context_->isEdge(), hasNullableCol_);
+        if (v.isStr()) {
+          auto iter = std::find_if(fields_.begin(), fields_.end(), [&col](const auto& field) {
+            return col == field.get_name();
+          });
```
Comment (yixinglu) [inline src/storage/exec/IndexOutputNode.h:340]: CHECK whether the iter is valid: `DCHECK(iter != xx.end());`

Comment (cangfengzhs) [reply src/storage/exec/IndexOutputNode.h:340]: > CHECK whether the iter is valid: `DCHECK(iter != xx.end());`

Just share the discussion details, it is not finished

**Comment by cangfengzhs** [comment]
This bug has been fixed in https://github.com/vesoft-inc/nebula/pull/3196

**Pull request closed without merging**